File: tests/nodejs/cases/case_092_reconcile_order.js
/**
 * Task case 092: reconcile_order
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case092ReconcileOrder(input) {
  // Task: reconcile order records and prepare transport-ready payload.
  const taskName = 'reconcile_order';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'order',
    stage: 'reconcile',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is appended to retained audit state, outliving function scope.
  retainedAudit.push(payload);
  return 'ok';
}

module.exports = {
  case092ReconcileOrder,
};

File: tests/python/cases/case_092_reconcile_order.py
"""
Task case 092: reconcile order payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_092_reconcile_order(input_data):
    # Task: reconcile order records and prepare transport-ready payload.
    task_name = 'reconcile_order'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'order',
        'stage': 'reconcile',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is appended to RETAINED_AUDIT, outliving function scope.
    RETAINED_AUDIT.append(payload)
    return 'ok'
